torch/testing/_internal/common_utils.py

```diff
@@ -1039,6 +1039,13 @@ def _compareScalars(self, a, b, *,
 
         return _compare_scalars_internal(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
 
+    # Construct assert messages basd on internal debug message and user provided message.
+    def _get_assert_msg(self, msg, debug_msg=None):
+        if msg is None:
+            return debug_msg
+        else:
+            return f"\n{msg}" if debug_msg is None else f"{debug_msg}\n{msg}"
```

Comment:
Why the newline ahead of msg in the first case?

I would have thought this would be something like:

```
@staticmethod
def _make_assert_msg(user_msg, library_msg):
  if user_msg is None:
    return library_msg
  if library_msg is None:
    return user_msg
  return f"{libary_msg}\n{user_msg}"
```

Reply:
in many case `library_msg` can be 2+ lines and it's hard to differentiate which part is library and which part is user. thus having a newline for user-only message makes it easier to identify.

Reply:
but the way you suggested is more consistent too, I will incorporate in the next follow up patch. thanks for the suggestion.